### PR TITLE
React week4 homework 

### PR DESCRIPTION
**Pull request by shpomp**

### CODE REVIEW

File: react/week4/homework/src/App.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      setLoading(false);
+    }, 2000);
+
+    return () => cleanUp(id);
+  }, [searchInput]);
+
+  return (
+    <div className="App">
+      <SearchContext.Provider value={{ searchInput, setSearchInput }}>
+        <SearchBar></SearchBar>
```
Comment (nuha90) [inline react/week4/homework/src/App.js:36]: `<SearchBar />`

File: react/week4/homework/src/App.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      </SearchContext.Provider>
+      {loading && <p>loading....</p>}
+      {!loading && (
+        <ResultContext.Provider value={{ users }}>
+          <UsersList></UsersList>
```
Comment (nuha90) [inline react/week4/homework/src/App.js:41]: `<UsersList />`

File: react/week4/homework/src/components/UsersList.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <>
+          {usersList.users.map((user) => {
+            const userInfo = {
+              name: user.login,
+              avatar: user.avatar_url,
+              url: user.html_url,
+            };
+            return (
+              <UserContext.Provider key={user.id} value={userInfo}>
+                <User></User>
```
Comment (nuha90) [inline react/week4/homework/src/components/UsersList.js:24]: `<User />`

**Review by nuha90 — COMMENTED** [review]
Good job 👍

File: react/week4/homework/src/App.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  useEffect(() => {
+    setLoading(true);
+    // debouncing
+    const id = setTimeout(async () => {
+      const result = await fetch(
+        `https://api.github.com/search/users?q=${searchInput}`
+      );
+      const fetchedData = await result.json();
+      result.ok ? setUsers(fetchedData.items) : setUsers([]);
```
Comment (nuha90) [inline react/week4/homework/src/App.js:26]: can you show the error to the user

File: react/week4/homework/src/App.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import React, { useEffect, useState } from "react";
+import SearchBar from "./components/SearchBar";
+import UsersList from "./components/UsersList";
+
+export const SearchContext = React.createContext();
+export const ResultContext = React.createContext();
+
+function App() {
+  const [loading, setLoading] = useState(true);
+  const [searchInput, setSearchInput] = useState("mojombo");
```
Comment (nuha90) [inline react/week4/homework/src/App.js:11]: why are you starting with "mojombo"?

Comment (shpomp) [reply react/week4/homework/src/App.js:11]: oh, it was just easier to have it as initial state - when making sure it is working, changing styling and etc.

**Comment by shpomp** [comment]
Thank you @nuha90  very much for your time and feedback ❤️  ☀️

**Pull request merged**